setup.py

```diff
@@ -0,0 +1,344 @@
+                rel_package = ext.name.split('.')[:-1]
+                package_data_d = os.path.join(self.build_lib, *(rel_package + ['.data']))
+
+                # Python files created/configured by CMake
+                src_py_dir = os.path.join(ext.cmake_install_prefix,
+                                          ext.cmake_install_python_files)
+                if os.path.isdir(src_py_dir):
+                    copy_tree(src_py_dir, self.build_lib)  # accepts existing dst dir
+                    shutil.rmtree(src_py_dir)  # avoid being collected to data dir
+
+                for d in ext.cmake_collect_dirs:
+                    log.info("  - Collecting %s (and everything under it)", d)
+                    src_dir = os.path.join(ext.cmake_install_prefix, d)
+                    dst_dir = os.path.join(package_data_d, d)
+                    if not os.path.isdir(dst_dir):
+                        shutil.copytree(src_dir, dst_dir)
+
+                log.info("==> Done building CMake project. Now building python extension\n")
+
+                # Make the temp include paths in the building the extension
+                ext.include_dirs += [os.path.join(self.build_temp, inc_dir)
+                                     for inc_dir in ext.include_dirs
+                                     if not os.path.isabs(inc_dir)]
+                ext.cmake_done = True
+
+        # Now build the extensions normally
+        build_ext.run(self, *args, **kw)
+
+    def _run_cmake(self, ext):
+        cmake = self._find_cmake()
+        cfg = 'Debug' if self.debug else 'Release'
+        self.outdir = os.path.abspath(ext.cmake_install_prefix)
+        log.info("Building lib to: %s", self.outdir)
+
+        cmake_args = [
+            # Generic options only. project options shall be passed as ext param
+            '-DCMAKE_INSTALL_PREFIX=' + self.outdir,
+            '-DPYTHON_EXECUTABLE=' + sys.executable,
+            '-DCMAKE_BUILD_TYPE=' + cfg,
+        ] + ext.cmake_flags + ["-D" + opt for opt in self.cmake_defs.split(",")]
+
+        if self.cmake_prefix:
+            cmake_args.append("-DCMAKE_PREFIX_PATH=" + self.cmake_prefix)
+
+        build_args = ['--config', cfg, '--', '-j4']  # , 'VERBOSE=1']
+
+        env = os.environ.copy()
+        env['CXXFLAGS'] = "{} -DVERSION_INFO='{}'".format(env.get('CXXFLAGS', ''),
+                                                          self.distribution.get_version())
+        if not os.path.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+
+        try:
+            subprocess.Popen("echo $CXX", shell=True, stdout=subprocess.PIPE)
+            log.info("[CMAKE] cmd: %s", " ".join([cmake, ext.sourcedir] + cmake_args))
+            subprocess.check_call([cmake, ext.sourcedir] + cmake_args,
+                                  cwd=self.build_temp, env=env)
+            subprocess.check_call(
+                [cmake, '--build', '.', '--target', 'install'] + build_args,
```

Comment:
I found it safer to have for cmake and cmake build runs to have the same environment, so I'd add `env=env` here too.

Reply:
good catch